addons/event_crm/tests/test_event_crm_flow.py

```diff
@@ -0,0 +1,184 @@
+        })
+        test_registrations += self.env['event.registration'].create({
+            'event_id': test_event.id,
+            'name': 'Test Registration 3',
+            'email': 'test2@test.com',
+            'phone': '0456000099',
+            'main_registration_id': test_registrations[0].id,
+        })
+
+        lead = self.env['crm.lead'].search([('event_id', '=', test_event.id)])
+
+        self.assertEqual(len(lead), 1,
+            "Event CRM: one lead sould be created for the set of attendees")
+
+        self.assertEqual(lead.registration_count, 2,
+            "Event CRM: registration which does not check the rule should not be linked to the lead")
+
+        self.assertEqual(len(lead.event_id.registration_ids), 3,
+            "Event CRM: three registrations should have been created for the event")
+
+        self.check_description(lead)
+
+    @mute_logger('odoo.addons.base.models.ir_model', 'odoo.models')
+    def test_event_crm_multi_rules(self):
+        # creation of a new event
+        test_event = self.env['event.event'].with_user(self.user_eventmanager).create({
+            'name': 'TestEvent',
+            'date_begin': datetime.datetime.now() + relativedelta(days=-1),
+            'date_end': datetime.datetime.now() + relativedelta(days=1),
+        })
+
+        test_lead_tag = self.env['crm.lead.tag'].create({'name': 'TagTest'})
+
+        # creation of rules
+        self.env['event.crm.rule'].with_user(self.user_eventmanager).create({
+            'name': 'TestEventCrmRule',
+            'lead_creation_basis': 'attendee',
+            'event_id': test_event.id,
+            'registration_filter_domain': [['email', 'ilike', '@test.com']],
+            'lead_tag_ids': test_lead_tag
+        })
+        self.env['event.crm.rule'].with_user(self.user_eventmanager).create({
+            'name': 'TestEventCrm',
+            'lead_creation_basis': 'order',
+            'event_id': test_event.id,
+            'lead_user_id': self.user_eventmanager.id,
+        })
+
+        # creation of three registrations for this event
+        test_registrations = self.env['event.registration'].create({
+            'event_id': test_event.id,
+            'name': 'Test Registration 1',
+            'email': 'test1@test.com',
+            'phone': '0456000042',
+        })
+        test_registrations += self.env['event.registration'].create({
+            'event_id': test_event.id,
+            'name': 'Test Registration 2',
+            'email': 'test2@other.com',
+            'phone': '0456000000',
+            'main_registration_id': test_registrations[0].id,
+        })
+        test_registrations += self.env['event.registration'].create({
+            'event_id': test_event.id,
+            'name': 'Test Registration 3',
+            'email': 'test2@test.com',
+            'phone': '0456000099',
+            'main_registration_id': test_registrations[0].id,
+        })
```

Comment:
You could factorise a bit and use the setUp() if TransactionalCase or even setupClass() with SavepointCase.
You can put there every variable you need that is common to every test you do (like test_event, test_registrations etc..). setup will be done before each test in TransactionalCase. setupClass will be done once before yours tests and transaction will be rollback after each test, keeping what is done in setupclass.

also, for test registrations you can do a create multi and do something like :
```
        Registration = self.env['event.registration']
        test_registrations = Registration.create({
            'event_id': test_event.id,
            'name': 'Main Test Registration',
            'email': 'main@test.com',
            'phone': '0456000042',
        })
        test_registrations |= Registration.create({
            'event_id': test_event.id,
            'name': 'Test Registration %s' % i,
            'email': 'test%s@test.com' % i,
            'phone': '045600009%s' % i,
            'main_registration_id': test_registrations[0].id,
        } for i in range(2))
```